library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/SessionReplayPrivacyTest.kt

```diff
@@ -15,282 +17,205 @@ import android.widget.NumberPicker
-        assertThat(mappers[3].mapper).isInstanceOf(CheckBoxMapper::class.java)
-        assertThat(mappers[4].type).isEqualTo(CheckedTextView::class.java)
-        assertThat(mappers[4].mapper).isInstanceOf(CheckedTextViewMapper::class.java)
-        assertThat(mappers[5].type).isEqualTo(Button::class.java)
-        assertThat(mappers[5].mapper).isInstanceOf(ButtonMapper::class.java)
-        assertThat(mappers[6].type).isEqualTo(EditText::class.java)
-        assertThat(mappers[6].mapper).isInstanceOf(EditTextViewMapper::class.java)
-        assertThat(mappers[7].type).isEqualTo(TextView::class.java)
-        assertThat(mappers[7].mapper).isInstanceOf(TextViewMapper::class.java)
-        assertThat(mappers[8].type).isEqualTo(ImageView::class.java)
-        assertThat(mappers[8].mapper).isInstanceOf(ViewScreenshotWireframeMapper::class.java)
+        for ((type, mapper) in actualMappers) {
+            val expectedMapper = expectedMappers.find { it.type == type }?.mapper
+            assertThat(mapper.javaClass.name).isEqualTo(expectedMapper?.javaClass?.name)
+        }
     }
 
-    @TestTargetApi(26)
-    @Test
-    fun `M return the AllowAll mappers W rule(MASK_USER_INPUT, above API 26)`() {
-        // When
-        val mappers = SessionReplayPrivacy.MASK_USER_INPUT.mappers()
+    // region Internal
 
-        // Then
-        assertThat(mappers.size).isEqualTo(9)
-        assertThat(mappers[0].type).isEqualTo(SeekBar::class.java)
-        assertThat(mappers[0].mapper).isInstanceOf(MaskAllSeekBarWireframeMapper::class.java)
-        assertThat(mappers[1].type).isEqualTo(SwitchCompat::class.java)
-        assertThat(mappers[1].mapper).isInstanceOf(MaskAllSwitchCompatMapper::class.java)
-        assertThat(mappers[2].type).isEqualTo(RadioButton::class.java)
-        assertThat(mappers[2].mapper).isInstanceOf(MaskAllRadioButtonMapper::class.java)
-        assertThat(mappers[3].type).isEqualTo(CheckBox::class.java)
-        assertThat(mappers[3].mapper).isInstanceOf(MaskAllCheckBoxMapper::class.java)
-        assertThat(mappers[4].type).isEqualTo(CheckedTextView::class.java)
-        assertThat(mappers[4].mapper).isInstanceOf(CheckedTextViewMapper::class.java)
-        assertThat(mappers[5].type).isEqualTo(Button::class.java)
-        assertThat(mappers[5].mapper).isInstanceOf(ButtonMapper::class.java)
-        assertThat(mappers[6].type).isEqualTo(EditText::class.java)
-        assertThat(mappers[6].mapper).isInstanceOf(EditTextViewMapper::class.java)
-        assertThat(mappers[7].type).isEqualTo(TextView::class.java)
-        assertThat(mappers[7].mapper).isInstanceOf(TextViewMapper::class.java)
-        assertThat(mappers[8].type).isEqualTo(ImageView::class.java)
-        assertThat(mappers[8].mapper).isInstanceOf(ViewScreenshotWireframeMapper::class.java)
+    private fun setApiLevel(newApiLevel: Int) {
+        Build.VERSION::class.java.setStaticValue("SDK_INT", newApiLevel)
```

Comment:
fyi: beware that these calls may have no effect at some point. Because after dozens of calls most likely JVM will inline this property value (because from JVM point of view it has the right to do so - it is `static final`), this is what we experienced in the past. Ideally in the future we shouldn't touch `static final` things with reflection.